frontend/src/components/Notes.tsx

```diff
@@ -0,0 +1,203 @@
+    if (confirm("Are you sure you want to delete this note?")) {
+      dispatch(deleteNote.request({ noteId: note.id, recipeId }))
+    }
+  }
+  return (
+    <div className={classNames("d-flex align-items-start", className)}>
+      <Avatar avatarURL={created_by.avatar_url} className="mr-2" />
+      <div className="w-100">
+        <p>
+          {created_by.email} |{" "}
+          {formatDistanceToNow(new Date(note.created), { addSuffix: true })}
+        </p>
+        {!isOpen ? (
+          <Markdown
+            className="cursor-pointer"
+            title="click to edit"
+            onClick={() => {
+              setEditing(true)
+            }}>
+            {note.text}
+          </Markdown>
+        ) : (
+          <>
+            <Textarea
+              autoFocus
+              className="my-textarea mb-2"
+              onKeyDown={e => {
+                if (e.key === "Escape") {
+                  onClose()
+                }
+                if (e.key === "Enter" && e.metaKey) {
+                  onCreate()
+                }
+              }}
+              minRows={5}
+              value={newText}
+              onChange={e => setNewText(e.target.value)}
+              placeholder="Add a note..."
+            />
+            {isOpen && (
+              <div className="d-flex justify-between align-center">
+                <div className="d-flex justify-between align-center">
+                  <ButtonPrimary
+                    size="small"
+                    onClick={onCreate}
+                    loading={updating}
+                    className="mr-2">
+                    save
+                  </ButtonPrimary>
+                  <ButtonSecondary size="small" onClick={onClose}>
+                    cancel
+                  </ButtonSecondary>
+                </div>
+                <ButtonSecondary size="small" onClick={onDelete}>
+                  delete
+                </ButtonSecondary>
+              </div>
+            )}
+          </>
+        )}
+      </div>
+    </div>
+  )
+}
+
+interface INoteCreatorProps {
+  readonly recipeId: number
+}
+function NoteCreator({ recipeId }: INoteCreatorProps) {
+  const dispatch = useDispatch()
+  const recipe = useSelector(state => {
+    const maybeRecipe = state.recipes.byId[recipeId]
+    if (!isSuccessLike(maybeRecipe)) {
+      return null
+    }
+    return maybeRecipe.data
+  })
+  if (!recipe) {
+    return null
+  }
+  const isOpen = recipe.creatingNewNote
+  const note = recipe.draftNote
+  const loading = recipe.addingNoteToRecipe
+
+  const onCreate = () => {
+    dispatch(addNoteToRecipe.request({ id: recipeId }))
+  }
+  const onCancel = () => {
+    dispatch(toggleCreatingNewNote({ recipeId, value: false }))
+  }
+  return (
+    <div>
+      <Textarea
+        id="new_note_textarea"
+        className={classNames({
+          "my-textarea mb-2": isOpen,
+          textarea: !isOpen
+        })}
+        onKeyDown={e => {
```

Comment:
I think it would be better to pull these anon functions into the body and name them

could even pull them out into a hook `useCreateNoteHandlers()`